niworkflows/anat/ants.py

```diff
@@ -613,6 +614,164 @@ def init_atropos_wf(name='atropos_wf',
+        Estimated peak memory consumption of the most hungry nodes
+    bids_suffix : str
+        Sequence type of the first input image. For a list of acceptable values see
+        https://bids-specification.readthedocs.io/en/latest/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#anatomy-imaging-data
+    atropos_refine : bool
+        Enables or disables the whole ATROPOS sub-workflow
+    atropos_use_random_seed : bool
+        Whether ATROPOS should generate a random seed based on the
+        system's clock
+    atropos_model : tuple or None
+        Allows to specify a particular segmentation model, overwriting
+        the defaults based on ``bids_suffix``
+    name : str, optional
+        Workflow name (default: ``'n4_only_wf'``).
+
+    Inputs
+    ------
+    in_files
+        List of input anatomical images to be bias corrected,
+        typically T1-weighted.
+        If a list of anatomical images is provided, subsequently
+        specified images are used during the segmentation process.
+        However, only the first image is used in the registration
+        of priors.
+        Our suggestion would be to specify the T1w as the first image.
+
+    Outputs
+    -------
+    out_file
+        :abbr:`INU (intensity non-uniformity)`-corrected ``in_files``
+    out_mask
+        Calculated brain mask
+    bias_corrected
+        Same as "out_file", provided for consistency with brain extraction
+    bias_image
+        The :abbr:`INU (intensity non-uniformity)` field estimated for each
+        input in ``in_files``
+    out_segm
+        Output segmentation by ATROPOS
+    out_tpms
+        Output :abbr:`TPMs (tissue probability maps)` by ATROPOS
+
+    """
+    wf = pe.Workflow(name)
+
+    inputnode = pe.Node(niu.IdentityInterface(fields=['in_files', 'in_mask']),
+                        name='inputnode')
+
+    outputnode = pe.Node(niu.IdentityInterface(
+        fields=['out_file', 'out_mask', 'bias_corrected', 'bias_image',
+                'out_segm', 'out_tpms']),
+        name='outputnode')
+
+    # Create brain mask
+    thr_brainmask = pe.Node(Binarize(thresh_low=2), name='binarize')
+
+    # INU correction
+    inu_n4_final = pe.MapNode(
+        N4BiasFieldCorrection(
+            dimension=3, save_bias=True, copy_header=True,
+            n_iterations=[50] * 5, convergence_threshold=1e-7, shrink_factor=4,
+            bspline_fitting_distance=200),
+        n_procs=omp_nthreads, name='inu_n4_final', iterfield=['input_image'])
+
+    # Check ANTs version
+    try:
+        inu_n4_final.inputs.rescale_intensities = True
+    except ValueError:
+        warn("The installed ANTs version too old. Please consider upgrading to "
+             "2.1.0 or greater.", DeprecationWarning)
+
+    wf.connect([
+        (inputnode, inu_n4_final, [('in_files', 'input_image')]),
+        (inputnode, thr_brainmask, [(('in_files', _pop), 'in_file')]),
+        (thr_brainmask, outputnode, [('out_mask', 'out_mask')]),
```

Comment:
Should the dilated mask be output, or the undilated one (which is what we're doing here)

Reply:
I'd say the undilated. I believe we should refactor the Binarize interface so that it can do the binary closing and opening, which will give a smoother mask (and that will be good for the dilated mask too).